include/bswap.h

```diff
@@ -42,6 +45,18 @@
 #endif
 #endif
 
+/* macOS */
+#ifdef __APPLE__
+#define __LITTLE_ENDIAN LITTLE_ENDIAN
+#define __BIG_ENDIAN BIG_ENDIAN
+#define __BYTE_ORDER BYTE_ORDER
+#if __BYTE_ORDER == BIG_ENDIAN
+#define __BIG_ENDIAN_BITFIELD
+#elif __BYTE_ORDER == LITTLE_ENDIAN
+#define __LITTLE_ENDIAN_BITFIELD
+#endif
+#endif /* __APPLE__ */
```

Comment:
Please do not invent new way for determining endiantity. We already use autoconf which has standard macros for it.

Reply:
It's how it works without reshuffling too much the chain of ifdef'ed code.
You have to consider that I'm not actually determining there endianness, only perusing how it is pre-defined in the macOS-specific way (i.e., as in machine/endian.h). Indeed the right side of the defines (no double under-score) comes from the Darwin header.

Reply:
Autoconf has macro AC_C_BIGENDIAN to check for endianity, see:
https://www.gnu.org/software/autoconf/manual/autoconf.html

There is absolutely no reason for adding new special ifdef for Apple systems as autoconf can handle it.

Reply:
Tried, but it does not work. The ifdef'ed HAVE_SYS_ISA_DEFS_H part is a bit non-standard and makes the following (ifdef APPLE) necessary. Again, consider that it is not determining endianness, only doing what you do under HAVE_SYS_ISA_DEFS_H (e.g., define the __*) which are not available anywhere on a Darwin system otherwise.

Reply:
But why to use HAVE_SYS_ISA_DEFS_H? WORDS_BIGENDIAN should be defined only on big endian systems.

Reply:
Now I did it in 0adadb61c88add40a42d312c50c8f3218412c28a